source/val/validate_conversion.cpp

```diff
@@ -46,8 +46,10 @@ spv_result_t ConversionPass(ValidationState_t& _, const Instruction* inst) {
                << "Expected input to be float scalar or vector: "
                << spvOpcodeString(opcode);
 
-      if (_.IsCooperativeMatrixType(result_type) ||
-          _.IsCooperativeMatrixType(input_type)) {
+      if (_.IsCooperativeMatrixNVType(result_type) ||
+          _.IsCooperativeMatrixNVType(input_type) ||
+          _.IsCooperativeMatrixKHRType(result_type) ||
+          _.IsCooperativeMatrixKHRType(input_type)) {
```

Comment:
Should this be:
```suggestion
      if ((_.IsCooperativeMatrixNVType(result_type) &&
           _.IsCooperativeMatrixNVType(input_type)) ||
          (_.IsCooperativeMatrixKHRType(result_type) &&
           _.IsCooperativeMatrixKHRType(input_type))) {
```

Several similar instances below.

Reply:
No. We should get into `if` part when either of input or result is cooperative matrix, because else part have only stub, while `if` part properly handles (inside CooperativeMatrixShapesMatch):

```
  if (m1_type->opcode() != m2_type->opcode()) {
    return diag(SPV_ERROR_INVALID_DATA, inst)
           << "Expected cooperative matrix types";
  }
```

effectively expecting both input and result are cooperative matrices.

Reply:
I guess my question is more: can you mix the NV and KHR types for conversions? If not, where is that checked?

Reply:
In the code I have mentioned, due to:

`    OpTypeCooperativeMatrixKHR = 4456,`

`    OpTypeCooperativeMatrixNV = 5358,`

Reply:
You cannot mix as per my knowledge.